### ISSUE TITLE
Removed bracket '(' in HTML an XML output

### ISSUE THREAD

**Issue by johgo17** [doxygen/doxygen#8556]
**Describe the bug**
If I run doxygen on the .hpp file, Doxygen removes from the function the '(' bracket between '>' and '>'. The Doxyfile and the source file are attached to reproduce the Bug. My Doxygen version is 1.8.17, but the same problem occurs under the latest version (1.9.2).

**Expected behavior**
The '(' shows up in the xml and html output.

**Screenshots**
The html Documentation (and also the xml) is missing the '(' bracket.
![Doxygen-html](https://user-images.githubusercontent.com/84388671/118651865-42711380-b7e6-11eb-83b0-3b0208a7dc00.png)
![source](https://user-images.githubusercontent.com/84388671/118651885-48ff8b00-b7e6-11eb-8f76-4b40260a0486.png)

**To Reproduce**
I run `doxygen Doxyfile` in the folder attached.
[dox_issue.zip](https://github.com/doxygen/doxygen/files/6501288/dox_issue.zip)

**Version**
Doxygen Version: 1.8.17
Ubuntu 20.04. on Linux 5.4.72-microsoft-standard-WSL2 

**Additional context**
For the xml output it is weird, that in the test_8hpp.xml file the brackets are there as expected.

**Comment by albert-github** [comment]
I see the missing opening round bracket in the HTML and XML output as well (for 1.8.17, 18.19, 1.9.1 and current master).
The place where the opening round bracket does show up is in de representation of the code lines on the other places the opening round bracket is missing, also in the XML output where I see:
```
        <type>std::function&lt; std::unique_ptr&lt; Test &gt;)&gt;</type>
        <definition>std::function&lt;std::unique_ptr&lt;Test&gt;)&gt; some_function</definition>
```

Note in the (Version) description of the issue you speak of version 1.8.19 but the image is from version 1.8.17 (I've corrected it. It has, in this case, no effect on the outcome though) .

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8557

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
